adam_preprocessing/shape_stroke_graph_inference.py

```diff
@@ -22,12 +23,19 @@
 from utils import accuracy, get_stroke_data, LinearModel, load_data, STRING_OBJECT_LABELS
 
 
-def update_features_yaml(features_yaml, *, predicted_object: str):
+def update_features_yaml(features_yaml, *, predicted_object: Sequence[str]):
     # jac: not terribly efficient to do a full deepcopy, but who cares, this should be a small dict
     # anyway... also this is probably much less expensive than the GNN inference itself.
     result = deepcopy(features_yaml)
     for object_ in result["objects"]:
-        object_["stroke_graph"]["concept_name"] = predicted_object
+        if len(predicted_object) == 1:
+            object_["stroke_graph"]["concept_name"] = predicted_object
```

Comment:
It should indeed! Thanks for catching that. Since the multiple decode experiment was the last thing I worked on, I never ran the script with only 1 decode per object (the default behavior)